catalyst-gateway/bin/src/service/common/objects/fragments_processing_summary.rs

```diff
@@ -0,0 +1,72 @@
+//! Define information about fragments that were processed.
+
+use poem_openapi::{types::Example, Enum, NewType, Object};
+
+#[derive(NewType)]
+#[oai(example = true)]
+/// Unique ID of a fragment.
+///
+/// A fragment is the binary representation of a signed transaction.
+/// The fragment ID is the hex-encoded representation of 32 bytes.
+pub(crate) struct FragmentId(String);
+
+impl Example for FragmentId {
+    fn example() -> Self {
+        Self("7db6f91f3c92c0aef7b3dd497e9ea275229d2ab4dba6a1b30ce6b32db9c9c3b2".into())
+    }
+}
+#[derive(Enum)]
+/// The reason for which a fragment was rejected.
+pub(crate) enum ReasonRejected {
+    /// This fragment was already processed by the node.
+    FragmentAlreadyInLog,
+    /// This fragment failed validation.
+    FragmentInvalid,
+    /// One of the previous fragments was rejected and `fail_fast` is enabled.
+    PreviousFragmentInvalid,
+    /// One of the mempools rejected this fragment due to reaching capacity limit.
+    PoolOverflow,
+}
+
+#[derive(Object)]
+#[oai(example = true)]
+/// Information about a rejected fragment.
+pub(crate) struct RejectedFragment {
+    #[oai(rename = "id")]
+    #[oai(validator(max_length = 64, min_length = 64, pattern = "[0-9a-f]{64}"))]
+    /// The ID of the rejected fragment.
+    fragment_id: FragmentId,
```

Comment:
FragmentId should start with he `0x` prefix

Reply:
I've added a note regarding adding the prefix in the future, currently, it'll stay as it is, in order to keep compatibility with the existing endpoint.